src/lib/components/fileUploader/fileUploader.js

```diff
@@ -0,0 +1,234 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import Uppy from "@uppy/core";
+import AwsS3 from "@uppy/aws-s3";
+import Webcam from "@uppy/webcam";
+import { DashboardModal, Dashboard } from "@uppy/react";
+import { checkValidations } from "../../utils/validation";
+import {
+  componentPropTypes,
+  defaultComponentPropTypes,
+} from "../../utils/componentPropTypes";
+
+import styles from "./fileUploader.module.scss";
+
+class FileUploader extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalOpen: false,
+      validationErrors: "",
+    };
+    this.uppy = Uppy({
+      restrictions: {
+        maxNumberOfFiles: props.maxNumberOfFiles,
+        allowMultipleUploads: props.allowMultipleUploads,
+        autoProceed: props.autoProceed,
+        maxFileSize: props.maxFileSize || undefined,
```

Comment:
Maybe we need to set a `default prop` here rather than `undefined`? if its just a number we could put, 2500, but we would need to display a message if it was greater than this number? or maybe `validatonError` already does that for us?

Reply:
ValidationError already does that for us, and with undefined we are saying there is no maxFileSize...in some cases for videos for example we don´t know which is going to be the max size